lms/djangoapps/course_api/v0/views.py

```diff
@@ -0,0 +1,451 @@
+
+        GET /api/courses/v0/{course_id}/content/
+
+        GET /api/courses/v0/{course_id}/content/
+
+    **Response Values**
+
+        * category: The type of content.
+
+        * due: The due date.
+
+        * uri: The URI to use to get details of the content entity.
+
+        * id: The unique identifier for the content entity.
+
+        * name: The name of the course.
+    """
+
+    def get(self, request, course_id):
+        """
+        GET /api/courses/v0/{course_id}/content/
+        """
+        depth = int(request.QUERY_PARAMS.get('depth', 1))
+        course_descriptor, course_key, _course_content = get_course(course_id, depth=depth)
+
+        if not course_descriptor:
+            raise Http404
+
+        response_data = []
+        for child in course_descriptor.get_children():
+            response_data.append(_serialize_content_with_children(request, course_key, child, depth - 1))
+
+        return Response(response_data)
+
+
+class CourseContentDetail(APIViewWithPermissions):
+    """
+    **Use Case**
+
+        CourseContentDetail returns a JSON collection for a specified
+        CourseContent entity. If the specified CourseContent is the Course, the
+        course representation is returned. You can use the uri values in the
+        children collection in the JSON response to get details for that content
+        entity.
+
+        The optional **depth** parameter that allows clients to get child content down to the specified tree level.
+
+    **Example Request**
+
+          GET /api/courses/v0/{course_id}/content/{content_id}/
+
+    **Response Values**
+
+        * category: The type of content.
+
+        * name: The name of the content entity.
+
+        * due:  The due date.
+
+        * uri: The URI of the content entity.
+
+        * id: The unique identifier for the course.
+
+        * children: Content entities that this content entity contains.
+    """
+
+    def get(self, request, course_id, content_id):
+        """
+        GET /api/courses/v0/{course_id}/content/{content_id}/
+        """
+        depth = int(request.QUERY_PARAMS.get('depth', 0))
+
+        # Verify the course exists.
+        course_descriptor, course_key, _course_content = get_course(course_id)
+        if not course_descriptor:
+            raise Http404
+
+        content_descriptor, _content_key, _content = get_course_child(request, request.user, course_key, content_id)
+
+        # TODO Add category filtering
+        response_data = _serialize_content_with_children(request, course_key, content_descriptor, depth)
+        return Response(response_data, status=status.HTTP_200_OK)
+
+
+class CourseList(PaginatedListAPIViewWithPermissions):
+    """
+    **Use Case**
+
+        CourseList returns paginated list of courses in the edX Platform. The list can be
+        filtered by course_id
+
+    **Example Request**
+
+          GET /api/courses/v0/
+          GET /api/courses/v0/?course_id={course_id1},{course_id2}
```

Comment:
It feels weird to me to specify multiple course ids this way in a parameter called course_id. What's the use case for this?